app/javascript/react/src/redux/welcome/greetingsSlice.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from "@reduxjs/toolkit";
+import fetchData from "./fetchAPi";
+
+const initialState = {
+  data: '',
+  isSuccess: false,
+  message: '',
+  loading: false
+};
+
+const randomMessage = createSlice({
+  name: 'RandomMessage',
+  initialState,
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchData.pending, (state) => {
+        state.loading = true;
+      })
+      .addCase(fetchData.fulfilled, (state, action) => {
+        state.loading = false;
+        // state.isSuccess = ture;
+        state.data = action.payload
```

Comment:
**OPTIONAL**- It is not good practice in redux to mutate the state of your application. In redux your state must be [immutable](https://reacttraining.com/blog/state-in-react-is-immutable). Please use the spread operator to modify the state of your application 👍 👍